jplag.endToEndTesting/src/main/java/de/jplag/end_to_end_testing/helper/FileHelper.java

```diff
@@ -0,0 +1,149 @@
+package de.jplag.end_to_end_testing.helper;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import de.jplag.end_to_end_testing.constants.TestDirectoryConstants;
+import de.jplag.options.LanguageOption;
+
+/**
+ * Helper class to perform all necessary operations or functions on files or folders.
+ */
+public class FileHelper {
+
+    private FileHelper() {
+        // private constructor to prevent instantiation
+    }
+
+    /**
+     * Merges all contained filenames together without extension
+     * @param files whose names are to be merged
+     * @return merged filenames
+     */
+    public static String getEnclosedFileNamesFromCollection(Collection<File> files) {
+        StringBuilder stringBuilder = new StringBuilder(files.size());
+        for (File file : files) {
+            String fileName = file.getName();
+            stringBuilder.append(fileName.substring(0, fileName.lastIndexOf('.')));
+        }
+
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Load all possible language is resource path
+     * @param directoryNames folder names for which the language options should be listed.
+     * @return list of all LanguageOptions included in the resource path
+     */
+    public static List<LanguageOption> getLanguageOptionsFromPath(String[] directoryNames) {
+        List<LanguageOption> returnList = new ArrayList<>();
+        for (String languageDirectoryName : directoryNames) {
+            returnList.add(LanguageOption.valueOf(languageDirectoryName));
+        }
+        return returnList;
```

Comment:
That's really helpful thanks! actually it shouldn't happen but you never know. especially when names change in the languages it could lead to what you described.